src/hooks/libp2p/useBootLibp2pNode.ts

```diff
@@ -0,0 +1,62 @@
+import { nodeConfig } from 'lib/config/node_config';
+import { createLibp2p, Libp2p } from 'libp2p';
+import { useCallback, useEffect } from 'react';
+import { setLibp2p } from 'store/app/actions';
+import { useDispatch, useSelector } from 'store/index';
+import { log } from '../../lib/utils/logger';
+import { useLibp2p } from './useLibp2p';
+
+/**
+ * This hook is responsible for initializing a libp2p node
+ * The browser uses this node to communicate with archaeologists
+ * who themselves are libp2p nodes
+ *
+ * Boot Sequence:
+ * 1. Initialize node with a pre-defined config
+ * 2. Start the node
+ * 3. Add event listeners to the node for peer discovery and connection
+ */
+
+export function useBootLibp2pNode() {
+  const dispatch = useDispatch();
+  const globalLibp2pNode = useSelector(s => s.appState.libp2pNode);
+  const { handlePublicKeyStream, onPeerConnect, onPeerDisconnect, onPeerDiscovery } = useLibp2p();
+
+  const createAndSetNode = useCallback(async (): Promise<Libp2p> => {
+    const newLibp2pNode = await createLibp2p(nodeConfig);
+
+    // set global libp2p node for later use
+    dispatch(setLibp2p(newLibp2pNode));
+
+    return newLibp2pNode;
+  }, [dispatch]);
+
+  const addNodeEventListeners = useCallback(
```

Comment:
Following up on the other conversation below... Adding the listeners to the `libp2pNode` is technically mutating state after the dispatch action has happened on line 29. It may be better to add the listeners before the dispatch action.

Reply:
For instance, I would add the listeners after create on line 26 and before dispatch on line 29

Reply:
@knoll3 -- great point -- will get this updated

Reply:
@knoll3 -- this change is updated. going to hold off on the removal of listeners in this branch, will possibly add in a separate branch -- want to discuss this further.